src/mbgl/style/style_parser.cpp

```diff
@@ -224,8 +224,10 @@ void StyleParser::parseSources(const JSValue& value) {
                 continue;
             }
 
-            // We always assume the default configuration for GeoJSON sources.
             info = std::make_unique<SourceInfo>();
```

Comment:
Move this line to the top of the `SourceType::GeoJSON` case, and then set `info->maxZoom = geojsonvt->options.maxZoom` right after `geojsonvt = ...`, so that the `if (geojsonvt)` conditional isn't necessary.